### PR TITLE
API in ModuleBuilder to declare entries in gitignore

### LINKED ISSUE #9716: New API in ModuleBuilder to declare entries in gitignore
Modules in custom-jhlite might need to add new content to `.gitignore`.

And current generated `.gitignore` contains content for all the things that jhlite is able to generate, it would be better for each module to declare things, so you get a file that fits what you have in your generated project.
A few examples : 
- `.svelte-kit/` -> only in svelte client
- `okta.sh`, `auth0.sh` -> only in you generate oauth2 with okta
- `.angular/` -> for angular client
- gradle wrapper
- maven wrapper
- ...

### PR DESCRIPTION
**Pull request by murdos**
Fixes #9716

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/module/domain/JHipsterModule.java [hunk 1]
```diff
@@ -654,6 +694,13 @@ public JHipsterModuleSpringFactoriesBuilder springTestFactories() {
       return springFactories.computeIfAbsent(SpringFactoryType.TEST_FACTORIES, key -> JHipsterModuleSpringFactories.builder(this));
     }
 
+    /**
+     * Add new entries to the {@code .gitignore} file.
+     */
+    public JHipsterModuleGitIgnoreBuilder gitIgnore() {
```
Comment (renanfranca) [inline src/main/java/tech/jhipster/lite/module/domain/JHipsterModule.java:700]: Interesting comment. This comment is useful when usign the jhipster-custom module to create a new module.

File: src/main/java/tech/jhipster/lite/module/domain/gitignore/GitIgnoreEntry.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      return value;
+    }
+  }
+
+  record GitIgnoreComment(String value) implements GitIgnoreEntry {
+    private static final String COMMENT_PREFIX = "#";
+
+    public GitIgnoreComment(String value) {
+      Assert.notBlank("value", value);
+      this.value = value.startsWith(COMMENT_PREFIX) ? value : COMMENT_PREFIX + " " + value;
```
Comment (renanfranca) [inline src/main/java/tech/jhipster/lite/module/domain/gitignore/GitIgnoreEntry.java:28]: Nice approach! Give the developer freedom to implement the comment in the way he/she wants 😎

**Review by renanfranca — APPROVED** [review]
Thank you for your work! That looks great to me 😃👍

File: src/main/java/tech/jhipster/lite/module/domain/JHipsterModule.java (hunk 1 above)
Comment (murdos) [reply src/main/java/tech/jhipster/lite/module/domain/JHipsterModule.java:700]: Yes, and also for contributors to new modules in jhipster-lite :)
See #3972

**Pull request merged**